<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>719</width>
    <height>604</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout_2">
    <item>
     <layout class="QVBoxLayout" name="verticalLayout">
      <property name="sizeConstraint">
       <enum>QLayout::SetMinimumSize</enum>
      </property>
      <item>
       <widget class="QLabel" name="label">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>Output stage:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QComboBox" name="stage_selector">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <item>
         <property name="text">
          <string>Input</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Filter</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Edges</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Contours</string>
         </property>
        </item>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>H min:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSpinBox" name="hmin_input">
        <property name="maximum">
         <number>180</number>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_3">
        <property name="text">
         <string>H max:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSpinBox" name="hmax_input">
        <property name="maximum">
         <number>180</number>
        </property>
        <property name="value">
         <number>180</number>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_4">
        <property name="text">
         <string>S min:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSpinBox" name="smin_input">
        <property name="maximum">
         <number>255</number>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_5">
        <property name="text">
         <string>S max:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSpinBox" name="smax_input">
        <property name="maximum">
         <number>255</number>
        </property>
        <property name="value">
         <number>255</number>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_8">
        <property name="text">
         <string>Canny threshold 1:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSpinBox" name="threshold1_input">
        <property name="maximum">
         <number>100</number>
        </property>
        <property name="value">
         <number>50</number>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_9">
        <property name="text">
         <string>Canny threshold 2:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSpinBox" name="threshold2_input">
        <property name="maximum">
         <number>100</number>
        </property>
        <property name="value">
         <number>50</number>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_6">
        <property name="text">
         <string>Minimal shape area:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSpinBox" name="minarea_input">
        <property name="maximum">
         <number>1000000000</number>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_7">
        <property name="text">
         <string>Curve approximation epsilon:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSpinBox" name="epsilon_input">
        <property name="maximum">
         <number>1000000000</number>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="verticalSpacer">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QLabel" name="display">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>719</width>
     <height>19</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="separator"/>
    <addaction name="actionOpen"/>
    <addaction name="actionSave"/>
    <addaction name="actionQuit"/>
   </widget>
   <addaction name="menuFile"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionOpen">
   <property name="text">
    <string>Open</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="text">
    <string>Save</string>
   </property>
  </action>
  <action name="actionQuit">
   <property name="text">
    <string>Quit</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
